docs/form-datasets.rst

```diff
@@ -138,29 +155,67 @@ The ``item`` blocks are analogous to rows in the CSV representation. Each ``item
 
-Expressions to look up values in datasets always start with ``instance("<instance name>")`` to identify which dataset is being accessed. If you have a choice list named ``places`` or an attached CSV with filename ``places.csv``, your lookup expressions will start with ``instance("places")``.
+In the example below, a user is first prompted to select a participant from the list of people found in an external file. Then, the selected participant's ``name`` is used to look up the ``place`` that participant is assigned to. A second dataset is attached from a ``places.csv`` file using :ref:`csv-external <form-datasets-attaching-csv>`. The assigned place is looked up by ``name`` and its ``label`` is displayed directly in a **note**.
 
-The next part of the expression is ``/root/item``. This means to consider every single **item** in your choice list or attached file.
+.. csv-table:: survey
+  :header: type, name, label, calculation
 
-You then generally need to have a filter in square brackets to specify which item(s) to actually use. For example, if you have a choice list or attached file named ``states`` and want to look up a value about a state that was entered by the user in a form field called ``my_state``, you would use ``name = ${my_state}`` as your filter expression. This is exactly the same kind of expression you would write as a ``choice_filter``.
+  select_one_from_file people, participant, Participant
+  calculate, assigned_place, , instance("people")/root/item[name=${participant}]/place
+  csv-external, places, , 
+  note, place_note, Assigned place: instance("places")/root/item[name=${assigned_place}]/label
 
-Your filter expression can result in one or more items being selected. Filtering to a result that includes multiple items is particularly useful for sums and counts. For example, to count the number of states with a population above a certain threshold:
+.. csv-table:: places.csv
+  :header: name, label, population, enumerator
 
-``count(instance("states")/root/item[population > ${pop_threshold}])``
+  c2139aae-5…, Ifedore, 270900, 6234
+  f1ad1a8a-c…, Magama, 311300, 2742
 
-To use a single value from the item(s) selected, specify which column/property to use after the filter. For example, if you want to look up the population of the state which your user filled in as ``my_state``, your full expression would be:
+.. csv-table:: people.csv
+  :header: name, label, place, visits, last_visit, phone
 
-``instance("states")/root/item[name = ${my_state}]/population``
+  da0ee575-d…, Xue (Ifedore) - 2341745, c2139aae-5…, 1, 2024-5-1, 2341745
+  c51c32ac-1…, Dia (Ifedore) - 9868545, c2139aae-5…, 3, 2024-4-12, 9868545
 
-To get the total population across states with a population above a certain threshold:
+The components of lookup expressions
```

Comment:
"Understanding lookup expressions" is more welcoming.

Reply:
How about just "lookup expressions"? We have a note right below saying it's ok to pattern match and I don't want to give the impression that you have to understand all the details.